test/Test.agda

```diff
@@ -165,7 +165,7 @@ sumMon (x ∷ xs) = x <> sumMon xs
 
 -- Instance argument proof obligation that should not turn into a class constraint
 hd : (xs : List a) → ⦃ NonEmpty xs ⦄ → a
```

Comment:
Is this certainly not gonna intervene with instances/typeclasses, specifically the rule "any instance argument that depends on another **visible** argument"?

Reply:
I don't understand, what will interfere with typeclasses? The rule is: any instance argument that does not depend on a visible argument is turned into a type class constraint. The constraints here are either not instance arguments (the tactic arguments to `error` and `undefined`) or depend on a visible argument (the constraints of `head` and friends).